prql-elixir/native/prql/src/lib.rs

```diff
@@ -0,0 +1,155 @@
+fn to_result_tuple(result: Result<String, prql_compiler::ErrorMessages>) -> NifResult<Response> {
+    match result {
+        Ok(sql) => Ok(Response {
+            status: atoms::ok(),
+            result: sql,
+        }),
+        Err(e) => Ok(Response {
+            status: atoms::error(),
+            result: e.to_json(),
+        }),
+    }
+}
+
+/// Get the dialect from an atom. By default `Generic` dialect will be used
+fn dialect_from_atom(a: Atom) -> prql_compiler::sql::Dialect {
+    use prql_compiler::sql::Dialect as D;
+
+    if a == atoms::ansi() {
+        D::Ansi
+    } else if a == atoms::bigquery() {
+        D::BigQuery
+    } else if a == atoms::clickhouse() {
+        D::ClickHouse
+    } else if a == atoms::generic() {
+        D::Generic
+    } else if a == atoms::hive() {
+        D::Hive
+    } else if a == atoms::mssql() {
+        D::MsSql
+    } else if a == atoms::mysql() {
+        D::MySql
+    } else if a == atoms::postgres() {
+        D::PostgreSql
+    } else if a == atoms::sqllite() {
+        D::SQLite
+    } else if a == atoms::snowflake() {
+        D::Snowflake
+    } else {
+        D::Generic
+    }
+}
+
+impl From<CompileOptions> for prql_compiler::sql::Options {
+    /// Get `prql_compiler::sql::Options` options from `CompileOptions`
+    fn from(o: CompileOptions) -> Self {
+        prql_compiler::sql::Options {
+            format: o.format,
+            dialect: Some(dialect_from_atom(o.dialect)),
+            signature_comment: o.signature_comment,
+        }
+    }
+}
+
+#[derive(Clone, NifStruct, Debug)]
+#[module = "PRQL.Native.CompileOptions"]
+pub struct CompileOptions {
+    /// Pass generated SQL string trough a formatter that splits it
+    /// into multiple lines and prettifies indentation and spacing.
+    ///
+    /// Defaults to true.
+    pub format: bool,
+
+    /// Target dialect you want to compile for.
+    ///
+    /// Because PRQL compiles to a subset of SQL, not all SQL features are
+    /// required for PRQL. This means that generic dialect may work with most
+    /// databases.
+    ///
+    /// If something does not work in dialect you need, please report it at
+    /// GitHub issues.
+    ///
+    /// If None is used, `sql_dialect` flag from query definition is used.
+    /// If it does not exist, [Dialect::Generic] is used.
+    pub dialect: Atom,
+
+    /// Emits the compiler signature as a comment after generated SQL
+    ///
+    /// Defaults to true.
+    pub signature_comment: bool,
+}
+
+#[derive(NifTuple)]
+pub struct Response {
+    /// status atom `:ok` or `:error`
+    status: Atom,
+
+    /// result string
+    result: String,
+}
+
+#[rustler::nif]
+/// compile a prql query into sql
+pub fn compile(prql_query: &str, options: CompileOptions) -> NifResult<Response> {
+    let opts = prql_compiler::sql::Options::from(options);
+    let result = prql_compiler::compile(prql_query, Some(opts));
+    to_result_tuple(result)
```

Comment:
FYI now you have the `From` implemented, this will even work:

```
    let result = prql_compiler::compile(prql_query, Some(options.into()));
    to_result_tuple(result)
```
FYI now you have the `From` implemented, this will even work (but the current one is also fine):

```suggestion
    let result = prql_compiler::compile(prql_query, Some(options.into()));
    to_result_tuple(result)
```
```

Reply:
Yeah its working, pushed